src/Raven.Server/Documents/Replication/ReplicationLoader.cs

```diff
@@ -747,6 +853,38 @@ private TcpConnectionInfo GetConnectionInfo(ReplicationNode node, bool external)
             return null;
         }
 
+        public X509Certificate2 GetCertificateForReplication(ReplicationNode node, out TcpConnectionHeaderMessage.AuthorizationInfo authorizationInfo)
+        {
+            authorizationInfo = null;
+            var exNode = node as ExternalReplication;
+            if (exNode == null)
+                return _server.Server.Certificate.Certificate;
+
+            if (exNode.PullReplicationAsEdgeSettings?.CertificateThumbprint != null)
+            {
+                using (_server.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+                using (context.OpenReadTransaction())
+                {
+                    var key = Constants.Certificates.FeaturePrefix + exNode.PullReplicationAsEdgeSettings.CertificateThumbprint;
```

Comment:
You have no way here to limit this for a single db. In particular, just knowing the certificate thumbprint allows you to use it from any db on the server, which is probably not ideal

Reply:
No, you don't.
When you update a pull replication task you can provide it with a certificate, that will be stored internally (encrypted) and used.
No reuse between pull replications